plugins/bundle/plugin.go

```diff
@@ -310,14 +312,17 @@ func (p *Plugin) UnregisterBulkListener(name interface{}) {
 
 // Config returns the plugins current configuration
 func (p *Plugin) Config() *Config {
+	p.cfgMtx.RLock()
+	defer p.cfgMtx.RUnlock()
```

Comment:
Can't races still occur if a different goroutine updates `p.config` between the call to `Config()` and usage of the returned pointer?

Reply:
IIUC you're saying there's a goroutine that's holding the read lock in `Config()` and another goroutine tries to update the config? In this case, the write call is blocked till the read is complete.

Reply:
One option here is just to return a copy of the config. Something like

```go
// Config returns the plugins current configuration
func (p *Plugin) Config() *Config {
	p.cfgMtx.RLock()
	defer p.cfgMtx.RUnlock()
	return &Config{
		Bundles: p.getBundlesCpy(),
	}
}
```
The other fields of the `Config` structure were deprecated 5 years ago, so I think we should be ok. WDYT @johanfylling?

Reply:
That works 👍 .
Another option could maybe be to make `p.config` a pointer `*Config`; so that when it's swapped, the reading goroutine will be operating against the old config it previously retrieved.

Reply:
Ok updated.